readme.md

```diff
@@ -125,2440 +274,51 @@ Add `Sig` to your Zig project using `build.zig.zon` file (available for Zig >= 0
-
-<br/>
-
-**Options**
-<br/>
+Zig's own definition: `Zig is a general-purpose programming language and toolchain for maintaining robust, optimal and reusable software.`
 
-```zig
-const GetBlocksOptions = struct {
-    commitment: ?types.Commitment = null,
-};
-```
+1. **Optimized performance**: Zig provides control over how your program runs at a low level, similar to languages like C. It allows fine-grained control over aspects such as memory management and system calls, which can lead to improved performance.
 
-**Usage**
-<br/>
+2. **Safety focus**: Zig has features built in to prevent common bugs and safety issues common in C. For example, it includes built-in testing and bounds checking, which can help avoid problems such as buffer overflows and undefined behavior.
 
-```zig
-const std = @import("std");
-const sig = @import("sig");
-const rpc = sig.rpc;
+3. **Readability and maintainability**: Zig syntax is designed to be straightforward and clear. This can make the code easier to understand, more maintainable, and less prone to bugs.
 
+4. **No hidden control flow**: Zig doesn't allow hidden control-flow like exceptions in some other languages. This can make it easier to reason about what your program is doing.
 
-const allocator = std.heap.page_allocator;
+5. **Integration with C**: Zig has excellent interoperation with C. You can directly include C libraries and headers in a Zig program, which can save time when using existing C libraries.
 
-pub fn main() !void {
-    var client = try rpc.Client.init(allocator, .{ .http_endpoint = HTTP_ENDPOINT });
-    defer client.deinit();
+6. **Custom allocators**: Zig allows you to define custom memory allocation strategies for different parts of your program. This provides the flexibility to optimize memory usage for specific use-cases or workloads.
 
-    var resp = try client.getBlocksWithLimit(400, 25, .{});
-    defer resp.deinit();
+### Note
 
-    if (resp.err()) |err| {
-        std.log.err("error response: {any}", .{err});
-        return;
-    }
+- Zig is still an evolving language.
+- Many of the low-level APIs have been stabilized but `std.http.Client` and `std.json` are still WIP targetting stable implementations by `>=0.11`.
+- Currently, `std.http.Client` [leaks](https://github.com/ziglang/zig/blob/447a30299073ce88b7b26d18d060a345beac5276/lib/std/http/Client.zig#L913) and is failing some tests, fix is in works.
+<br><br>
 
-    std.log.debugf("blocks: {any}", .{resp.result()});
-}
-```
+## 🧩 Modules
 
-</details>
+- **[Gossip](src/gossip)** - A gossip spy node, run by: `sig gossip` or `zig build run -- gossip`
 
-<details>
-<summary><code>getBlockTime</code> - Returns the estimated production time of a block</summary>
-<br/>
+- **[Core](src/core)** - Core data structures shared across modules
 
-**Params:** <code>(slot: u64)</code>
+- **[RPC Client](src/rpc)** - A fully featured HTTP RPC client with ability to query all on-chain data along with sending transactions
```

Comment:
probs want to also link the `docs/api.md` somewhere here too

Reply:
probs wana also include a link/section for `Contributing`

Reply:
I included a section in the readme called Develop that describes running tests and benchmarks, and it links to the Setup section and CONTRIBUTING.md. Do you think there's a better way to organize this?